### ISSUE TITLE
Launcher icon indicates 1 unread message although everything is marked as read

### ISSUE THREAD

**Issue by hebu** [WhisperSystems/TextSecure#2358]
Since the last update, the app icon is indicating 1 unread message (by displaying the red badge on the shortcut icon), although all messages are read. Also, when a new message is received, the badge remains "1". I did not test with 2 unread messages.

The interesting part in the log file seems to be this line:

    W/ShortcutBadger( 5874): No support for: com.sec.android.app.launcher

Device info from the log file:

> Device : samsung GT-I9305 (m3xx)
> Android: JSS15J.I9305XXUEMKC
> OS Host: SEP-108
> App    : TextSecure 2.4.1

**Comment by McLoo** [comment]
:) the log entry will always be there. see https://github.com/WhisperSystems/ShortcutBadger/blob/d4ee33747d09b6167907105d197bb42f33de13b6/ShortcutBadger/src/main/java/me/leolin/shortcutbadger/ShortcutBadger.java#L79

Do you have some more log file content?

**Comment by hebu** [comment]
Sure. Attached is the latest log, after marking all as read from the context menu.


Am 15. Januar 2015 23:58:29 MEZ, schrieb McLoo <notifications@github.com>:
>:) the log entry will always be there. see
>https://github.com/WhisperSystems/ShortcutBadger/blob/d4ee33747d09b6167907105d197bb42f33de13b6/ShortcutBadger/src/main/java/me/leolin/shortcutbadger/ShortcutBadger.java#L79
>
>Do you have some more log file content? 
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/WhisperSystems/TextSecure/issues/2358#issuecomment-70178952
Device : samsung GT-I9305 (m3xx)
Android: JSS15J.I9305XXUEMKC
OS Host: SEP-108
App    : TextSecure 2.4.1

--------- beginning of /dev/log/main
W/KeyCachingService( 5874): Incrementing activity count...
W/KeyCachingService( 5874): Decrementing activity count...
D/AbsListView( 5874): Get MotionRecognitionManager
W/DirectoryRefreshListener( 5874): Scheduling for: 1421366746729
D/AbsListView( 5874): onVisibilityChanged() is called, visibility : 4
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onVisibilityChanged() is called, visibility : 0
D/AbsListView( 5874): unregisterIRListener() is called 
W/KeyCachingService( 5874): Incrementing activity count...
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): Get MotionRecognitionManager
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onDetachedFromWindow
D/AbsListView( 5874): unregisterIRListener() is called 
E/ViewRootImpl( 5874): sendUserActionEvent() mView == null
W/KeyCachingService( 5874): Decrementing activity count...
D/AbsListView( 5874): Get MotionRecognitionManager
D/AbsListView( 5874): onVisibilityChanged() is called, visibility : 4
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onVisibilityChanged() is called, visibility : 0
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
W/KeyCachingService( 5874): Incrementing activity count...
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onVisibilityChanged() is called, visibility : 4
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onDetachedFromWindow
D/AbsListView( 5874): Get MotionRecognitionManager
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
W/KeyCachingService( 5874): Decrementing activity count...
D/AbsListView( 5874): unregisterIRListener() is called 
D/Scrubber( 5874): scrubbing input
D/dalvikvm( 5874): JIT code cache reset in 10 ms (1048536 bytes 1/0)
D/dalvikvm( 5874): GC_CONCURRENT freed 1332K, 10% free 26833K/29812K, paused 8ms+27ms, total 225ms
D/AbsListView( 5874): [unregisterDoubleTapMotionListener]
I/MotionRecognitionManager( 5874):   .unregisterListener : / listener count = 0->0,  
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onVisibilityChanged() is called, visibility : 4
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onVisibilityChanged() is called, visibility : 0
D/AbsListView( 5874): unregisterIRListener() is called 
W/KeyCachingService( 5874): Incrementing activity count...
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onDetachedFromWindow
D/AbsListView( 5874): Get MotionRecognitionManager
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
W/KeyCachingService( 5874): Decrementing activity count...
D/AbsListView( 5874): onVisibilityChanged() is called, visibility : 0
D/AbsListView( 5874): unregisterIRListener() is called 
W/KeyCachingService( 5874): Incrementing activity count...
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onDetachedFromWindow
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onDetachedFromWindow
D/AbsListView( 5874): unregisterIRListener() is called 
E/ViewRootImpl( 5874): sendUserActionEvent() mView == null
W/MmsSmsDatabase( 5874): Executing query: SELECT _id, body, read, type, address, address_device_id, subject, thread_id, status, date_sent, date_received, m_type, msg_box, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (read = 0) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (read = 0) ORDER BY date_received ASC)
W/ShortcutBadger( 5874): No support for: com.sec.android.app.launcher
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onDetachedFromWindow
D/AbsListView( 5874): unregisterIRListener() is called 
E/ViewRootImpl( 5874): sendUserActionEvent() mView == null
W/KeyCachingService( 5874): Decrementing activity count...
D/AbsListView( 5874): Get MotionRecognitionManager
D/AbsListView( 5874): onVisibilityChanged() is called, visibility : 4
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onVisibilityChanged() is called, visibility : 0
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
W/KeyCachingService( 5874): Incrementing activity count...
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onVisibilityChanged() is called, visibility : 4
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): onDetachedFromWindow
D/AbsListView( 5874): Get MotionRecognitionManager
D/AbsListView( 5874): unregisterIRListener() is called 
D/AbsListView( 5874): unregisterIRListener() is called 
W/KeyCachingService( 5874): Decrementing activity count...
D/AbsListView( 5874): unregisterIRListener() is called

**Comment by jeremymasters** [comment]
Latest version of Textsecure is 2.4.2 but you're running 2.4.1. Don't nt know if that matters for this issue though.

**Comment by hebu** [comment]
I updated and rebooted the phone, but the issue is still there.

Device : samsung GT-I9305 (m3xx)
Android: JSS15J.I9305XXUEMKC
OS Host: SEP-108
App    : TextSecure 2.4.2

--------- beginning of /dev/log/main
W/PartProvider( 4822): Got master secret: null
D/dalvikvm( 4822): GC_CONCURRENT freed 515K, 16% free 10311K/12260K, paused 6ms+22ms, total 55ms
D/dalvikvm( 4822): GC_CONCURRENT freed 606K, 17% free 10724K/12768K, paused 3ms+2ms, total 57ms
I/dalvikvm( 4822): Could not find method android.provider.Telephony$Sms.getDefaultSmsPackage, referenced from method org.thoughtcrime.securesms.util.Util.isDefaultSmsProvider
W/dalvikvm( 4822): VFY: unable to resolve static method 1272: Landroid/provider/Telephony$Sms;.getDefaultSmsPackage (Landroid/content/Context;)Ljava/lang/String;
D/dalvikvm( 4822): VFY: replacing opcode 0x71 at 0x000a
W/DirectoryRefreshListener( 4822): Scheduling for: 1421409946792
W/KeyCachingService( 4822): Incrementing activity count...
W/KeyCachingService( 4822): Decrementing activity count...
D/dalvikvm( 4822): GC_FOR_ALLOC freed 646K, 16% free 11236K/13324K, paused 26ms, total 26ms
D/libEGL  ( 4822): loaded /system/lib/egl/libEGL_mali.so
D/libEGL  ( 4822): loaded /system/lib/egl/libGLESv1_CM_mali.so
D/libEGL  ( 4822): loaded /system/lib/egl/libGLESv2_mali.so
E/        ( 4822): Device driver API match
E/        ( 4822): Device driver API version: 23
E/        ( 4822): User space API version: 23 
E/        ( 4822): mali: REVISION=Linux-r3p2-01rel3 BUILD_DATE=Wed Oct 30 09:36:10 KST 2013 
D/OpenGLRenderer( 4822): Enabling debug mode 0
W/service ( 4822): Broadcasting new secret...
D/dalvikvm( 4822): GC_EXPLICIT freed 288K, 14% free 12613K/14596K, paused 3ms+6ms, total 85ms
W/MmsSmsDatabase( 4822): Executing query: SELECT _id, body, read, type, address, address_device_id, subject, thread_id, status, date_sent, date_received, m_type, msg_box, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (read = 0) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (read = 0) ORDER BY date_received ASC)
W/SendReceiveService( 4822): Got a MasterSecret broadcast...
W/PartProvider( 4822): Got master secret: org.thoughtcrime.securesms.crypto.MasterSecret@42a478c8
W/ShortcutBadger( 4822): No support for: com.sec.android.app.launcher
W/KeyCachingService( 4822): Incrementing activity count...
W/KeyCachingService( 4822): Decrementing activity count...
D/AbsListView( 4822): Get MotionRecognitionManager
W/DirectoryRefreshListener( 4822): Scheduling for: 1421409946792
D/AbsListView( 4822): onVisibilityChanged() is called, visibility : 4
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): onVisibilityChanged() is called, visibility : 0
D/AbsListView( 4822): unregisterIRListener() is called 
W/KeyCachingService( 4822): Incrementing activity count...
D/AbsListView( 4822): unregisterIRListener() is called 
W/RecipientProvider( 4822): Cache miss [ASYNC]!
D/dalvikvm( 4822): GC_CONCURRENT freed 264K, 12% free 13279K/15048K, paused 5ms+3ms, total 36ms
D/dalvikvm( 4822): WAIT_FOR_CONCURRENT_GC blocked 20ms
D/dalvikvm( 4822): WAIT_FOR_CONCURRENT_GC blocked 18ms
W/CursorWrapperInner( 4822): Cursor finalized without prior close()
W/RecipientProvider( 4822): Cache miss [ASYNC]!
W/RecipientProvider( 4822): Cache miss [ASYNC]!
W/RecipientProvider( 4822): Cache miss [ASYNC]!
W/RecipientProvider( 4822): Cache miss [ASYNC]!
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
W/RecipientProvider( 4822): Cache miss [ASYNC]!
W/RecipientProvider( 4822): Cache miss [ASYNC]!
W/RecipientProvider( 4822): Cache miss [ASYNC]!
W/RecipientProvider( 4822): Cache miss [ASYNC]!
D/AbsListView( 4822): unregisterIRListener() is called 
W/BitmapUtil( 4822): creating scaled bitmap with scale 0.5625 => 1152x720
D/dalvikvm( 4822): GC_FOR_ALLOC freed 217K, 12% free 13519K/15244K, paused 42ms, total 43ms
I/dalvikvm-heap( 4822): Grow heap (frag case) to 24.664MB for 10485776-byte allocation
D/dalvikvm( 4822): GC_FOR_ALLOC freed 1K, 7% free 23758K/25488K, paused 34ms, total 34ms
W/Emoji   ( 4822): bitmap for this page was null
W/Emoji   ( 4822): bitmap for this page was null
W/Emoji   ( 4822): bitmap for this page was null
D/dalvikvm( 4822): GC_CONCURRENT freed 19K, 7% free 23743K/25488K, paused 12ms+5ms, total 66ms
D/AbsListView( 4822): unregisterIRListener() is called 
W/BitmapUtil( 4822): rough scale 2048x1280 => 2048x1280
W/BitmapUtil( 4822): fine scale 2048x1280 => 1152x720
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/dalvikvm( 4822): GC_FOR_ALLOC freed 10431K, 42% free 17221K/29276K, paused 33ms, total 33ms
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): Get MotionRecognitionManager
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): onDetachedFromWindow
D/AbsListView( 4822): unregisterIRListener() is called 
E/ViewRootImpl( 4822): sendUserActionEvent() mView == null
W/MmsSmsDatabase( 4822): Executing query: SELECT _id, body, read, type, address, address_device_id, subject, thread_id, status, date_sent, date_received, m_type, msg_box, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (read = 0) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (read = 0) ORDER BY date_received ASC)
W/ShortcutBadger( 4822): No support for: com.sec.android.app.launcher
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): onDetachedFromWindow
D/AbsListView( 4822): unregisterIRListener() is called 
E/ViewRootImpl( 4822): sendUserActionEvent() mView == null
W/KeyCachingService( 4822): Decrementing activity count...
D/AbsListView( 4822): Get MotionRecognitionManager
D/AbsListView( 4822): onVisibilityChanged() is called, visibility : 4
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): onVisibilityChanged() is called, visibility : 0
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
W/KeyCachingService( 4822): Incrementing activity count...
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): onVisibilityChanged() is called, visibility : 4
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): onDetachedFromWindow
D/AbsListView( 4822): Get MotionRecognitionManager
D/AbsListView( 4822): unregisterIRListener() is called 
D/AbsListView( 4822): unregisterIRListener() is called 
W/KeyCachingService( 4822): Decrementing activity count...
D/AbsListView( 4822): unregisterIRListener() is called

**Comment by Wydarr** [comment]
I can confirm the behavior on TS 2.4.2 running on a Galaxy Note 3.
The log file is here: https://gist.github.com/anonymous/ed9a2f6d5943a91e87d0
Tried uninstalling, clearing data all the lot. No luck.

**Comment by hebu** [comment]
Should it not be possible to show the true number of unread messages on the affected devices, it would already be helpful, if it just wouldn't show *any* number of unread messages.

**Comment by McLoo** [comment]
@mcginty as you have samungs devices around... could be vendor (-api) specific

**Comment by hebu** [comment]
So, would it be an option to detect a Samsung device and just remove the badge from the icon?

**Comment by jeremymasters** [comment]
My Galaxy S4 behaves correctly with the unread messages count, so blanketing this as all samsung may not be completely correct.

**Comment by eX00r** [comment]
I can confirm the same behavior in Samsung Galxy SII (i900) with latest Cyanogenmod installed. 
Keeps indicating several new Messages from one of my contacts from time to time.

**Comment by fabsmusicjunkie** [comment]
i have the same issue with my moto g (lte), stock android. you need a log?

**Comment by Wydarr** [comment]
The behavior is the same at version 2.5.1. 
Not to mention the fact that the "rebranding" is unfortunate to say the least. For example the main icon us blue, but when entering the unlocking password the icon is still green. 
So on one hand I have a bug that stands for 20 days now and instead of fixing it we have a... frigging rebranding. 
Oh well...

**Comment by mcginty** [comment]
@Wydarr please only leave comments in issues if they provide new information that can help identify the root cause of bugs.

If you want the issues fixed faster, we accept pull requests or productive diagnostic info.

**Comment by Wydarr** [comment]
Off topic: By all means, apologies if I offended anyone. 
On the other hand my comment provided the  (basic) info that the behavior still stands in the latest version, albeit spiced with a little bit of venting. 
The root cause (of the venting of course) is that I am a very early adopter of Text Secure, I have campaigned vigorously among my acquaintances for its exclusive use (in my phone it completely replaces the standard messaging app) and since the problems appeared, three weeks ago the app feels (at least from my point of view) less... composed. So once again apologies if egos were bruised. 

On topic: Here is the updated log for the latest version:
https://gist.github.com/anonymous/c6d877b6cf76a7eaa7d4

P.S. Alas I have no programming skills but let me know the ways in which I can help you with the bug and I by all means I shall.

**Comment by McLoo** [comment]
@eX00r you don't have the unread count on the app icon but as a notification, right? CM launcher has no support for icon badges.
@hebu do you also have a notification fr unread messages? Or just the count on the icon?

**Comment by eX00r** [comment]
Yes, only in the notification bar of android.
Am 11.02.2015 09:16 schrieb "McLoo" <notifications@github.com>:

**Comment by hebu** [comment]
I have the icon badge only, no notification.

**Comment by McLoo** [comment]
hey guys, would you mind to answer how you mark the messages as unread.

- by tapping the notification to open the conversation (or by opening TS from drawer or task list)
- by expanding the notification and tap "Mark (**all**) as read"
- by tapping "Mark all read" from the conversation list menu inside TS

The last method is the only thread and read-state independent and should set *every* message unread.
The first two are are set by thread id and only if read != 0.

**Comment by fabsmusicjunkie** [comment]
for me, none of this options work.
I already tried to delete the whole conversation, but this also doesn't make a difference

**Comment by hebu** [comment]
Hello!

Usually the first, sometimes the second option - and the last option only once for testing. Also, the counter stays at "1", no matter how many unread messages there are.

Greets,
HeBu

**Comment by McLoo** [comment]
@moxie0 meanwhile I suspect PushDatabase as the cause for all those phantom notifications
(@hebu's case still seems Samsung specific)
Regarding badge count: [pending push messages get added to the notification count](https://github.com/WhisperSystems/TextSecure/blob/e8b947dfde020212bfaa54b8e2ff547236989011/src/org/thoughtcrime/securesms/notifications/MessageNotifier.java#L150)

There is a [one time batch](https://github.com/WhisperSystems/TextSecure/blob/f092e85b6223bb70121e597782d47f99431841dc/src/org/thoughtcrime/securesms/DatabaseUpgradeActivity.java#L204) that *should* queue decryption for pending messages.
And the [deletion of a handled message ](https://github.com/WhisperSystems/TextSecure/blob/d3271f548c7b5e3d7866268c4976f6f4a73beef7/src/org/thoughtcrime/securesms/jobs/PushDecryptJob.java#L77) - this is the only place I found where PushDatabase entries get deleted.

But there could be entries in the push table, that get never handled. If they were not queued on DB upgrade properly (bad timing of incoming message) or if there was an issue while the decrypt job was running... 
Unlikely, but not impossible?!

related issues: #2221, #2287

**Comment by zmanian** [comment]
@McLoo 

My observations

* No observed correlation with a user interaction for clearing the unread notification. This affected 2 of many notifications of from TextSecure.

* No user interactions causes the persistent notifications to clear.

* Notifications are only persistent when the on device message DB is locked.  When unlocked, notification is as expected. When locked, the two zombie unread notifications immediately appear.

**Comment by eX00r** [comment]
Hi,

usually I mark it as unread

   - by tapping "Mark all read" from the conversation list menu inside TS

But that is what I am doing sometimes just because of the flase
Notification about a unread message.
The fals notification is reproducable by forcing TS to block access without
password. In the second I press "Textsecure sperren" the false notification
appears.

2015-02-19 20:18 GMT+01:00 Zaki Manian <notifications@github.com>:

**Comment by kskumar** [comment]
I have the same issue on my Nexus 5 (LRX22C, rooted, TS 2.5.3). Phantom notification appears for the same contact when the phone is locked, whether through timeout or manually. After entering passphrase, the phantom notification disappears, but re-appears after lock.

@McLoo I believe you are correct about the unhandled entries in the push table. I pulled the messages.db from my phone and looked at the push table using SQLite Manager (Firefox extension). The table had one row, and the sender was the contact for whom I get the phantom notification. I removed that row and pushed the DB back to my phone. Boom, issue gone.

**Comment by McLoo** [comment]
@kskumar great! Thanks for checking! 
Do you know when the issue happened the first time? After an update or crash maybe?

**Comment by kskumar** [comment]
@McLoo I haven't had a TS crash in ages :). I'd say the issue appeared for me 2-3 updates ago. Not sure if it helps, but the message's timestamp (in the DB) indicates it is from Feb 14.

**Comment by McLoo** [comment]
@hebu do you have notifications disabled?

**Comment by hebu** [comment]
I have notifications enabled and I do not get false notifications. My issue and the original issue of this post is, that the launcher icon keeps showing 1 unread message in the icon badge, no matter how many read or unread messages exist.

**Comment by sebex** [comment]
Same here, TextSecure icon shows one unread message. When I saw it for the first time, I thought that it might be some kind of new feature, cool. But inside TS I didn't see any unread messages. I did use 'Mark all as read' just in case, but it didn't help... so then I selected all my threads and deleted all of them at once. The icon still showed one unread message. I also went to the Settings > Application manager > TextSecure, and pressed the 'Clear data' after which I had to register my number again... anyway the icon still shows one unread message. Notification bar is fine, it's just the icon.

Photo of the icon: https://imgur.com/7oDLT4e

TextSecure version: 2.5.3
Device: Samsung Galaxy GT-I9100
Android version: 4.1.2

**Comment by agrajaghh** [comment]
If the "1" is appearing after 'clear data', its not a buggy message or database I guess. Looks like a problem with `ShortcutBadger`.

@sebex Are you running the Galaxy S2 stock rom or some rooted custom rom?

**Comment by agrajaghh** [comment]
Version 1.0.2 is currently beeing used by TS, 1.0.3 got released today but it doesn't look to me like a bug causing this got fixed....

@mcginty @moxie0 It seems in v1.0.2 the permissions `INSTALL_SHORTCUT` and `UNINSTALL_SHORTCUT` are missing and some guys got a warning of terms violation (or even banned) from google play store. See https://github.com/leolin310148/ShortcutBadger/issues/14 and https://github.com/leolin310148/ShortcutBadger/issues/12 . So i guess its best to update to 1.0.3

**Comment by hebu** [comment]
May I cite ny original post where i found this in my logfile;

<quote>
The interesting part in the log file seems to be this line:

W/ShortcutBadger( 5874): No support for: com.sec.android.app.launcher
</quote>

Maybe the shortcut badger of the Galaxy S2 and S3 is just not supported?

**Comment by sebex** [comment]
@agrajaghh My Galaxy S2 is running the normal stock rom, it's not rooted.

**Comment by McLoo** [comment]
@hebu see https://github.com/WhisperSystems/TextSecure/issues/2358#issuecomment-70178952
(And sorry for messing up your issue!) 

@agrajaghh I think the permissions  are only needed if you mess with the stock launcher icons, which was disabled.

**Comment by hebu** [comment]
@McLoo ah yes, I see, sorry. But maybe you could improve the logging in this method and then we can provide new information to this issue?

**Comment by agrajaghh** [comment]
@sebex after using 'clear data' could you restart your phone and see if the '1' is still there?

@McLoo ah, I see...

@hebu could you try uninstalling TS and reinstalling it again? Or 'clear data' and restart the phone. (I know its a lot to ask, since currently just plaintext backup is working)

**Comment by McLoo** [comment]
@moxie0 maybe the PR according to this info for the fix the Samsung issues: https://github.com/leolin310148/ShortcutBadger/issues/10#issuecomment-78454929

**Comment by McLoo** [comment]
@moxie0 this is very likely to be fixed finally with PR https://github.com/WhisperSystems/ShortcutBadger/pull/4
according to https://github.com/leolin310148/ShortcutBadger/issues/21

**Comment by kskumar** [comment]
@McLoo a couple questions...

1. Would that also fix it for non-Samsung devices, such as Nexus 5? I ask this because other issues related to ghost "regular" notifications (as opposed to launcher icon badges) on non-Samsung devices (ex: #2221) have been closed as duplicates of this one.

2. I imagine said fix will prevent new messages from getting stuck in the push table. But will it also handle the ones that are already stuck there?

**Comment by hebu** [comment]
Should it already be fixed in TextSecure 2.12.0? Because I still have the "1" badge (and I did not try to reinstall yet) 

Device  : Samsung GT-I9305 / Android : 4.3

**Comment by agrajaghh** [comment]
@hebu nope, you have to wait until https://github.com/WhisperSystems/ShortcutBadger/pull/4 is in and ShortcutBadger is updated in TS

**Comment by McLoo** [comment]
@kskumar #2287 and #2221 are fixed with https://github.com/WhisperSystems/TextSecure/commit/ 
It should be in the upcoming 2.13.x 
This includes a fix for existing goats (omg keeping this typo)

**Comment by jeremymasters** [comment]
@McLoo Goats!

**Comment by hebu** [comment]
@McLoo Great! :)

**Comment by fabsmusicjunkie** [comment]
The error disappeared for me. Thanks a lot

**Comment by rhodey** [comment]
thanks for checking back in @fabsmusicjunkie :)

is anyone else still experiencing this bug? If I can get confirmation from two more people I think it'd be safe to close this until we hear otherwise.

**Comment by hebu** [comment]
For me it's fixed. Thanks to everyone involved fixing it!

**Comment by volcano99** [comment]
Fixed here too

**Comment by McLoo** [comment]
Just to clarify for documentation:
This issue was caused by ShortcutBadger not handling Samsung badge intents correctly. 
Fixed with ShortcutBadger 1.1.0 WS-Edition, in https://github.com/WhisperSystems/TextSecure/commit/a21e38e3349a5b1bdae0b0c485e66bb76260f25a

**Comment by rhodey** [comment]
thanks all :D

**Comment by typerhack** [comment]
Any result with this topic? Is there a way to fix the badge counter ? I have the same problem with my 2015 galaxy a7

**Comment by john-wilkinson** [comment]
Just ran across this same problem with new Moto Z2 play

Debug log:

https://gist.github.com/99ba457b351d4ff594354fe2f44c43b2
